src/gamestate/system_state.cpp

```diff
@@ -290,7 +290,7 @@ void state::on_key_down(virtual_key keycode, key_modifiers mod) {
 				ui::show_main_menu(*this);
 			}
 
-			map_state.on_key_down(keycode, mod);
+			map_state.on_key_down(keycode, mod, can_move_map_while_visible);
```

Comment:
similarly, instead of passing it as a parameter here, you could test the variable and just not call the function at all when you don't want keypresses to be sent to the map